src/x509.cpp

```diff
@@ -78,6 +78,17 @@ void X509Certificate::VerificationContext::validityCheck() const
     }
 }
 
+X509Certificate::VerificationContext&
+X509Certificate::VerificationContext::setVerificationCheckTime(Asn1Time checkTime)
+{
+    try {
+        _verificationCheckTime = std::chrono::system_clock::to_time_t(checkTime.toTimePoint());
```

Comment:
I think this is a good time to add a toUnixtime() function to the Asn1Time object. Where we also do the range management and let it throw in case that it doesn't fit.